### ISSUE TITLE
Persisting data and saving Query

### ISSUE THREAD

**Issue by Jesse-jApps** [smandaric/bigquerylayers#21]
Thanks for the awesome Plugin!

I have two little questions/suggestions?
1. Instead of iterating over every row of the query result and saving it as a csv file, why not use the bigquery capability to save the result as dataframe (`.to_dataframe()`) and then using geopandas to save everything as gpkg? 
2. In case the data in a BQ-table changes, would it be possible to save the query itself (maybe within the qgislayer), so that it could easily be updated?

**Comment by idejuan** [comment]
I agree this is an amazing plugin, so first of all my deepest gratitude to the creator and contributors. 
I fully support the suggestions you put in this issue, specially the need for saving the query itself so it can be updated easily. Thank you!!
